modules/datastore/src/Plugin/DkanApiDocs/DatastoreApiDocs.php

```diff
@@ -0,0 +1,376 @@
+   *   A configuration array containing information about the plugin instance.
+   * @param string $pluginId
+   *   The plugin_id for the plugin instance.
+   * @param mixed $pluginDefinition
+   *   The plugin implementation definition.
+   *
+   * @return static
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $pluginId,
+    $pluginDefinition
+  ) {
+    return new static(
+      $configuration,
+      $pluginId,
+      $pluginDefinition,
+      $container->get('module_handler'),
+      $container->get('string_translation'),
+      $container->get('dkan.metastore.service'),
+      $container->get('dkan.datastore.import_info')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function spec() {
+    $spec = $this->getDoc('datastore');
+    $querySchema = self::filterJsonSchemaUnsupported($this->replaceRefs($this->getDoc('datastore', 'query')));
+
+    // Reformat definitions.
+    foreach ($querySchema["definitions"] as $key => $def) {
+      $schemaName = "datastoreQuery" . ucfirst($key);
+      $def["title"] = "Datastore Query: " . (isset($def["title"]) ? $def["title"] : $key);
+      $spec["components"]["schemas"][$schemaName] = $def;
+    }
+    unset($querySchema["definitions"]);
+    $spec["components"]["schemas"]["datastoreQuery"] = $querySchema;
+
+    // Requirements are slightly different if resource is present in path.
+    $resourceQuerySchema = $this->resourceQueryAlter($querySchema);
+    $spec["components"]["schemas"]["datastoreResourceQuery"] = $resourceQuerySchema;
+
+    // Fill in examples.
+    $spec = $this->setUpExamples($spec);
+    // Convert json schema to params.
+    $spec = $this->setUpGetParameters($spec);
+
+    return $spec;
+  }
+
+  /**
+   * Set up examples throughout spec from real data.
+   *
+   * @param array $spec
+   *   Openapi spec.
+   *
+   * @return array
+   *   Modified spec.
+   */
+  private function setUpExamples(array $spec) {
+    $exampleIds = $this->getExampleIdentifiers();
+    $spec["components"]["parameters"]["datastoreDistributionUuid"]["example"] = $exampleIds['distribution'];
+    $spec["paths"]["/api/1/datastore/query"]["post"]["requestBody"]["content"]["application/json"]["example"]
+      = $this->queryExample($exampleIds['distribution']);
+    $spec["paths"]["/api/1/datastore/query/download"]["post"]["requestBody"]["content"]["application/json"]["example"]
+      = $this->queryExample($exampleIds['distribution'], "csv");
+    $spec["paths"]["/api/1/datastore/query/{identifier}"]["post"]["requestBody"]["content"]["application/json"]["example"]
+      = $this->queryExample();
+
+    $spec['components']['parameters']['datastoreUuid']["example"] = $exampleIds['resource'];
+
+    $spec["paths"]["/api/1/datastore/sql"]["get"]["parameters"][0]["example"] = $this->sqlQueryExample($exampleIds['distribution']);
+    return $spec;
+  }
+
+  /**
+   * Set up the GET parameters for datastore queries. WIP.
```

Comment:
Yes, these get parameters are very hard to document because we are really just passing a converted JSON object as a query string, which doesn't really map to any of the OpenAPI options for documenting query parameters. I want to take a second pass at documenting these well after this gets merged.